lib/Authentication/AuthTokens/ShibAuthToken.php

```diff
@@ -96,73 +97,67 @@ private function getAttributesInitToken(){
-                 $HTML .= "<div style='text-align: center;'>";
-                 $HTML .= '<a href="'.htmlspecialchars(\Factory::$properties['LOGOUTURL']).'"><b><font colour="red">Logout</font></b></a>';
-                 $HTML .= "</div>";
-                 echo ($HTML);
-                 die();
+        $configService = \Factory::getConfigService();
+        $identityProviders = $configService->getIdentityProvidersInfo();
+
+        foreach ($identityProviders as $provider) {
+            if ($provider['idp'] === $idp) {
+                $name = $provider['name'];
+                $helpUrl = $provider['help_url'];
+
+                if (empty($_SERVER['voPersonID'])) {
+                    die(
+                        "Did not receive required attributes from the "
+                        . "IDP $name to complete authentication. "
+                        . "Please contact gocdb-admins."
+                    );
+                }
+
+                if (empty($_SERVER['entitlement'])) {
+                    die(
+                        "Did not receive the required entitlement "
+                        . "attribute from the IDP $name. "
+                        . "Please contact gocdb-admins."
+                    );
+                }
+
+                if (!empty($provider['required_groups'])) {
+                    $entitlementValues = explode(';', $_SERVER['entitlement']);
+
+                    if (
+                        !array_intersect(
+                            $entitlementValues,
+                            $provider['required_groups']
+                        )
+                    ) {
+                        $HTML = "<ul>"
+                            . "<li>Login requires a GOCDB entitlement value "
+                            . "which was not provided for the IDP $name.</li>"
+                            . "<li>Please see here for more information: "
+                            . "<a href='$helpUrl' target='_blank'>"
+                            . "$helpUrl</a>.</li>"
+                            . "<li>Logout or restart your browser"
+                            . "and attempt to login again using an IDP "
+                            . "that provides a GOCDB entitlement.</li>"
+                            . "</ul>";
+                        $HTML .= "<div style='text-align: center;'>";
+                        $HTML .= "<a href=\""
+                            . htmlspecialchars(\Factory::$properties['LOGOUTURL'])
+                            . "\"><b><font color=\"red\">Logout</font></b></a>";
+                        $HTML .= "</div>";
+                        echo ($HTML);
+                        die();
+                    }
+                }
+
+                $this->principal = $_SERVER['voPersonID'];
+                $this->userDetails = [
+                    'AuthenticationRealm' => $provider['authenticationRealms']
```

Comment:
Given https://github.com/GOCDB/gocdb/pull/524/files#r1842006878, this will need a slight tweak

Reply:
Done in 01f3cd379b62c4b3119a25d09f7cef0d852b9904